bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -454,21 +454,51 @@ impl Height {
     /// assert_eq!(lock_time.to_consensus_u32(), n_lock_time);
     #[inline]
     pub fn to_consensus_u32(self) -> u32 { self.0 }
-}
 
-impl_parse_str_from_int_fallible!(Height, u32, from_consensus, Error);
+    /*
+    fn parse<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, ParseError> {
+        let n = s.as_ref().parse::<i64>().map_err(ParseError::invalid_int(s))?
+        let n = u32::try_from(n).map_err(|_| ParseError::Conversion(n))?;
+        Self::from_consensus(n).map_err(Into::into)
+    }
+    */
+}
 
 impl fmt::Display for Height {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
 }
 
 impl FromHexStr for Height {
-    type Error = Error;
+    type Error = ParseHeightError;
 
     #[inline]
     fn from_hex_str_no_prefix<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, Self::Error> {
-        let height = crate::parse::hex_u32(s)?;
-        Self::from_consensus(height)
+        parse_hex(s, Self::from_consensus)
+    }
+}
+
+impl_parse_str!(Height, ParseHeightError, parser(Height::from_consensus));
+
+/// Error returned when parsing block height fails.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ParseHeightError(ParseError);
+
+impl fmt::Display for ParseHeightError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.display(f, "block height", 0, LOCK_TIME_THRESHOLD - 1)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for ParseHeightError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        self.0.source()
+    }
+}
+
+impl From<ParseError> for ParseHeightError {
+    fn from(value: ParseError) -> Self {
+        Self(value)
```

Comment:
I'll try to run fmt and see if it makes diff too crazy or not.

Reply:
This whole not being able to format PRs because of one devs requirement to not use nightly, which is now required for clippy anyways is really a PITA.

Reply:
clippy warnings aren't too common and are easy to fix in the reasonably-rare case they show up in CI. Vs formatting issues which show up in probably 98% of all PRs unless the user is running a format tool locally. So the clippy nightly requirement isn't really a burden on individual developers.

The real issue is the Rust ecosystem's total lack of a stable working formatting tool. And even the unstable one is unable to format diffs. I hardly blame individual developers who are uncomfortable frequently downloading binary blobs from the developers of that mess.